File: app/src/main/java/com/jx/androiddemo/tool/ToastUtil.java
package com.jx.androiddemo.tool;

import android.annotation.SuppressLint;
import android.widget.Toast;

import com.jx.androiddemo.BaseApplication;

public class ToastUtil {
    private static Toast mToast;
    private static long mLastThreadId;

    //显示文本的Toast
    @SuppressLint("ShowToast")
    public static void showTextToast(String message) {
        if (null == mToast || Thread.currentThread().getId() != mLastThreadId) {
            mToast = Toast.makeText(BaseApplication.getInstance().getApplicationContext(), message, Toast.LENGTH_SHORT);
            mLastThreadId = Thread.currentThread().getId();
        }
        mToast.show();
    }
}

File: app/src/main/java/com/jx/androiddemo/tool/ToastUtil.kt
package com.jx.androiddemo.tool

import android.content.Context
import android.util.TypedValue
import android.view.Gravity
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast

object ToastUtil {
    private var toast: Toast? = null

    fun showToast(context: Context?, msg: String?) {
        if (null == msg) {
            return
        }
        try {
            toast?.cancel()
        } catch (e: Exception) {
        }
        toast = Toast.makeText(context, "", Toast.LENGTH_SHORT)
        toast?.setGravity(Gravity.CENTER, 0, 0)
        val v = toast?.getView()
        if (v is LinearLayout) {
            val v2 = v.getChildAt(0)
            if (v2 is TextView) {
                v2.setTextSize(TypedValue.COMPLEX_UNIT_SP, 16f)
            }
        }
        toast?.setText(msg)
        toast?.show()
    }
}